x/txfees/keeper/fees.go

```diff
@@ -152,19 +150,27 @@ func (k Keeper) swapFeeToBaseDenom(
 	}
 
 	// Swap the coin to base denom
-	var (
-		tokenOutAmount = sdk.ZeroInt() // Token amount in base denom
-		route          = []poolmanagertypes.SwapAmountInRoute{{
-			PoolId:        feetoken.PoolID,
-			TokenOutDenom: baseDenom,
-		}}
-	)
-	err = osmoutils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
-		tokenOutAmount, err = k.poolManager.RouteExactAmountIn(ctx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
-		return err
-	})
+	route := []poolmanagertypes.SwapAmountInRoute{{
+		PoolId:        feetoken.PoolID,
+		TokenOutDenom: baseDenom,
+	}}
+
+	cacheCtx, write := osmoutils.NoEventCacheContext(ctx)
+	tokenOutAmount, err := k.poolManager.RouteExactAmountIn(cacheCtx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
 	if err != nil {
-		return nil, sdk.Coins{takerFeeCoin}, nil
+		return nil, nil, fmt.Errorf("swap fee token: %w", err)
```

Comment:
I think returning an error here defeats the purpose of ApplyFuncIfNoError. we should prob just return `nil, coin, nil`

Reply:
why so? what should the caller do with unpaid fees if it receives an error due to some problems in pools while swapping?

Reply:
the idea is that the called just throws fees in x/txfees and does not worry about what happens under the hood. if there is a swapping err, then tokens silently go to the community pool.

Reply:
I'm not sure why it's mutually exclusive.

also by this logic you can say that that it makes sense to put `ApplyFuncIfNoError` only on the top level functions which I think is wrong.

The point is to allow non-critical actions to not fail the charge fee.
 
Every function can decide what's critical or not critical for them in order to function based on the function context. 

the point here is **not** to fail the entire chargeFee process if you can't swap.

Reply:
will fix